lib/vfscore/automount.c

```diff
@@ -346,10 +350,86 @@ extern const char vfscore_einitrd_start[];
 extern const char vfscore_einitrd_end;
 #endif /* CONFIG_LIBVFSCORE_AUTOMOUNT_EINITRD */
 
+#define LIBVFSCORE_EXTRACTOPT_MOUNT_RAMFS		(0x1 << 0)
+#define LIBVFSCORE_EXTRACTOPT_MOUNT_IFNOMOUNT		(0x1 << 1)
+#define LIBVFSCORE_EXTRACTOPT_FORMAT_AUTO		(0x1 << 2)
+
+static inline int vfscore_extract_parse_opts(const char *opts)
+{
+	char strbuf[64];
+	size_t cpylen;
+	const char *arg, *iter;
+	size_t arglen;
+	ssize_t match;
+	int optflags = 0x0;
+
+	for (iter = opts; iter != NULL; ) {
+		arg = iter;
+		arglen = uk_nextarg_r(&iter, LIBVFSCORE_MOUNTOPTS_SEP);
+
+		if (arglen == 0 && arg)
+			continue; /* just an empty arg */
+
+		/*
+		 * Option: "ramfs"
+		 * Values (ramfs=VAL):
+		 *    0 - Do not mount an underlying ramfs
+		 *    1 - Always mount an underlying ramfs
+		 *        (default if present without value)
+		 *    2 - Only mount underlying ramfs if no mount point exists
+		 */
+		match = uk_strnkeycmp(arg, arglen, "ramfs", "=");
+		if (match == 0) { /* only "ramfs" */
+			uk_pr_debug("fstab: Option for \"extract\": \"ramfs=1\" (always mount)\n");
+			optflags |= LIBVFSCORE_EXTRACTOPT_MOUNT_RAMFS;
+			optflags &= ~LIBVFSCORE_EXTRACTOPT_MOUNT_IFNOMOUNT;
+		} else if (match > 0) { /* "ramfs=N" */
+			int param;
```

Comment:
Even though it is not required in the above scope, maybe still declare it in the beginning with the others?

Reply:
I had a discussion with @andreittr regarding this and I think makes more sense to move such declaration to a subscopes if possible. It is still declared at the beginning of a scope.